pkg/platform/runtime/setup/setup.go

```diff
@@ -968,3 +976,20 @@ func (s *Setup) handleEvent(ev events.Eventer) error {
 	}
 	return nil
 }
+
+func (s *Setup) deleteOutdatedArtifacts(setup Setuper, changedArtifacts artifact.ArtifactChangeset, alreadyInstalled store.StoredArtifactMap) error {
+	storedArtifacts, err := s.store.Artifacts()
+	if err != nil {
+		return locale.WrapError(err, "err_stored_artifacts")
+	}
+
+	err = setup.DeleteOutdatedArtifacts(changedArtifacts, storedArtifacts, alreadyInstalled)
+	if err != nil {
+		multilog.Error("Could not delete outdated artifacts: %v, falling back to removing everything", err)
+		err = os.RemoveAll(s.store.InstallPath())
+		if err != nil {
+			return errs.Wrap(err, "Failed to clean installation path")
+		}
```

Comment:
```suggestion
		// This multilog is technically redundant and may be dropped after we can collect data on this error for a while as rollbar is not surfacing the returned error
		multilog.Error("Could not delete outdated artifacts: %s", errs.JoinMessage(err))
		return errs.Wrap(err, "Could not delete outdated artifacts")
```

This code feels really weird. We shouldn't be using such a fallback IMO. Looking at the code I don't see any reason why DeleteOutdatedArtifacts would fail whereas RemoveAll would succeed, it makes no sense. But looking at rollbar I can't find any records of "Failed to clean installation path".

There are a total of 80 occurrences of the "Could not delete outdated artifacts" error over 6 months. I feel like we're somehow loosing the "Failed to clean installation path" error on rollbar rather than that it's not occurring, so I'm inclined to say we should drop this RemoveAll and address the actual error, but seeing as we might be missing it I would suggest dropping the RemoveAll but keeping the multilog here and adding an `errs.JoinMessage(err)` so we have a better idea of why it failed. Odds are this is user input.